src/fstests/fstests.jinja2

```diff
@@ -63,7 +103,32 @@ class Job(BaseJob):
         if self._check_kvm_xfstest_env():
             if self._kernel_config(src_path) == 'pass':
                 if self._kernel_build(src_path) == 'pass':
-                    return self._run_smoke_tests()
-        else:
-            return 'fail'
+                    if self._run_smoke_tests() == 'pass':
+                        if self._get_xml_from_vm() == 'pass':
+                            return self._parse_fstests_result()
+        return 'fail'
+
+    def _submit_fstests(self, parent, child_nodes, db):
+        #hack to remove sublevel and name because they are null
+        del parent['revision']['version']['sublevel']
+        del parent['revision']['version']['name']
+        base_node = {
+            'kind': 'node',
+            'parent': parent['_id'],
+            'status': 'complete',
+            'revision': parent['revision'],
+            'path': parent['path']
+        }
+
+        for node in child_nodes:
+            base_node.update({
+                'name': node['name'],
+                'result': node['result'],
+            })
+            sent_node = db.submit({'node': base_node})[0]
+
+    def _submit(self, results, node_id, db):
+        node = super()._submit(results[0]['result'], node_id, db)
+        self._submit_fstests(node, results, db)
+        return node
```

Comment:
I think you can drop that and just focus on parsing the results in the current PR.  Otherwise, the changes to submit the results are going to delay when we can merge the code to parse the results.  That's why there are 2 issues in the backlog, first to parse the results and then to submit them to the API.

Reply:
ok, updated accordingly. It makes totally sense, 1 commit = 1 significant change.